### ISSUE TITLE
WebGL interpolation on Android devices

### ISSUE THREAD

**Issue by mike-000** [openlayers/openlayers#13500]
**Describe the bug**
WebGL interpolation behaves differently on Android and Windows devices.

**To Reproduce**
Run https://codesandbox.io/s/webgl-sea-level-forked-bsd855 on Windows and Android devices (I tried Android 12 and 4.4, Chrome and Opera, both give similar results).
On Windows the "sea" has a smooth interpolated edge.
![image](https://user-images.githubusercontent.com/49240900/159714726-1bb6ee20-70a7-434b-aafc-aab637523555.png)
On Android at some sea levels and locations the edge may be repeated or jagged
![image](https://user-images.githubusercontent.com/49240900/159715105-8b7b9ade-291a-4599-9335-5d3921ac00d2.png)
This does not appear to be related to pixel ratio as setting the map pixel ratio to 1 does make any noticeable difference.  The output from code used in rendering tests such as `webgl-data-tile-interpolate-true`  and `source-raster-webgl` does appear as expected when run on the same Android devices.

**Expected behavior**
Some differences between hardware might be expected but this seems more basic as both new and quite old Android devices produce the same unexpected result.

**Comment by mike-000** [comment]
The problem seems related to the calculation of elevation from RGB values in the style.  If the RGB tiles are converted to Float32 data tiles the interpolation produces the same smooth edge as in Windows https://codesandbox.io/s/webgl-sea-level-forked-bkyl0f. To handle all scenarios, such as `ReprojTile` and `gutter` a convenience method for `ol/source/TileImage` which returns promise of a canvas which can be used to populate the data tile might be desirable.

**Comment by tschaub** [comment]
Was this fixed by #13505?

**Comment by mike-000** [comment]
> Was this fixed by #13505?

No, but #13505 will allow the data tile method described above to be used on all devices if necessary, albeit without interpolation on older devices which do not support `OES_texture_float_linear` (similar to the canvas Sea Level example).

I have only checked this with Windows and Android - is anyone able to check https://bsd855,csb.app produces a smooth edge on any other systems, and if not is it fixed in https://bkyl0f,csb.app?

**Comment by mike-000** [comment]
Checks on BrowserStack and LambdaTest show this affects almost all mobile devices, both Android and IOS.  It seems to be a precision issue despite `highp` being used - with DEM tiles it typically occurs close a change in the green cbannel.

Float32 data tiles do not have this issue but for maximum device support they cannot be used unconditionally - as observed in #13529 many Samsung and Huawei devices do not support 'OES_texture_float', and many more do not support 'OES_texture_float_linear'.  The application would need to check for support and use data tiles where possible, falling back to RGB where not.
